compiler/cli/src/org/jetbrains/kotlin/cli/jvm/compiler/CompileEnvironmentUtil.java

```diff
@@ -88,6 +89,7 @@ private static void doWriteToJar(
             }
             if (includeRuntime) {
                 writeRuntimeToJar(stream, resetJarTimestamps);
+                writeReflectToJar(stream, resetJarTimestamps);
```

Comment:
As I mentioned in [KT-17344](https://youtrack.jetbrains.com/issue/KT-17344), I think we shouldn't pack kotlin-reflect if `-no-reflect` is specified. Having kotlin-reflect in the classpath can change the program behavior quite visibly, for example `toString` on lambdas will behave differently.

Could you take a look into it? If you do, please don't forget to add a test for the `-no-reflect` case. Otherwise I suppose it can be fixed separately

Reply:
I hope I haven't forgotten to deal with ```-no-reflect```. Now it seems better to fix  KT-43220 & KT-17344 separately. So I have excluded the commit(a3feef6)  from this PR and keep it simple.